### PR TITLE
Rework force_bindless_texel_buffer workaround

### PR DESCRIPTION
**Pull request by doitsujin**
Passes all DXBC tests, but will need DXIL support. Haven't come up with a new name yet.

### CODE REVIEW

File: libs/vkd3d-shader/spirv.c [hunk 1]
```diff
@@ -5685,13 +5685,20 @@ static void vkd3d_dxbc_compiler_emit_offset_buffer(struct vkd3d_dxbc_compiler *c
...
-    vkd3d_spirv_build_op_decorate1(builder, array_id, SpvDecorationArrayStride, 8);
+    member_ids[0] = vkd3d_spirv_get_type_id(builder, VKD3D_TYPE_UINT, 2);
+    member_ids[1] = member_ids[0];
+
+    struct_id = vkd3d_spirv_build_op_type_struct(builder, member_ids, ARRAY_SIZE(member_ids));
+    vkd3d_spirv_build_op_member_decorate1(builder, struct_id, 0, SpvDecorationOffset, 0);
+    vkd3d_spirv_build_op_member_decorate1(builder, struct_id, 1, SpvDecorationOffset, 8);
+
+    array_id = vkd3d_spirv_build_op_type_runtime_array(builder, struct_id);
+    vkd3d_spirv_build_op_decorate1(builder, array_id, SpvDecorationArrayStride, 16);
```
Comment (HansKristian-Work) [inline libs/vkd3d-shader/spirv.c:5701]: This commit should be squashed I think so that we don't completely break bisects.

**Review by HansKristian-Work — COMMENTED** [review]
Seems reasonable enough. Should be verified against the new test I made a PR for.

File: libs/vkd3d-shader/spirv.c (hunk 1 above)
Comment (doitsujin) [reply libs/vkd3d-shader/spirv.c:5701]: Will do once we get DXIL support, which then should also be squashed into this.

**Comment by doitsujin** [comment]
Closing because this doesn't work.

**Pull request closed without merging**